src/libaktualizr/uptane/uptane_init_test.cc

```diff
@@ -261,6 +261,44 @@ TEST(Uptane, InitializeFail) {
   }
 }
 
+/**
+ * Verifies if the system hostname is used as a primary ECU hardware ID
+ * if it's not specified in the configuration
+ *
+ * Checks actions:
+ *
+ * - [x] Use the system hostname as hardware ID if one is not provided
+ */
+TEST(Uptane, HostnameAsHardwareID) {
+  TemporaryDirectory temp_dir;
+  Config conf("tests/config/basic.toml");
+  conf.storage.path = temp_dir.Path();
+
+  boost::filesystem::copy_file("tests/test_data/cred.zip", temp_dir.Path() / "cred.zip");
+  conf.provision.provision_path = temp_dir.Path() / "cred.zip";
+
+  {
+    auto storage = INvStorage::newStorage(conf.storage);
+    auto http = std::make_shared<HttpFake>(temp_dir.Path());
+    KeyManager keys(storage, conf.keymanagerConfig());
+    Initializer initializer(conf.provision, storage, http, keys, {});
+    EXPECT_TRUE(initializer.isSuccessful());
+
+    EcuSerials ecu_serials;
+    EXPECT_TRUE(storage->loadEcuSerials(&ecu_serials));
+    EXPECT_GE(ecu_serials.size(), 1);
+
+    // A second element of the first tuple in ECU Serials tuple array is a primary hardware ID.
+    // Each client of the storage class needs to know this information.
+    // If it changes then corresponding changes should be done in each storage client.
+    // perhaps it makes sense to introduce get/setPrimaryHardwareID method and incapsulate
+    // this tech info within storage (or maybe some other entity)
```

Comment:
I grant that this isn't well-documented, but it is guaranteed in the current implementation that the first entry returned from `loadEcuSerials` is the primary. It's our convention, but also ensured by the SQL command: `SELECT serial, hardware_id FROM ecu_serials ORDER BY is_primary DESC;`. I'm not opposed to a shortcut getter as a convenience function for getting the primary hardware ID, but a setter seems unnecessary for now.

Reply:
ok